### PR TITLE
Add autocomplete warning about user input

### PR DESCRIPTION
**Pull request by rdrescher909**
## Summary

Adds a warning that users may elect to ignore autocomplete suggestions.

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by olijeffers0n — APPROVED** [review]

**Review by Ender2K89 — APPROVED** [review]

**Comment by JDJGInc** [comment]
Looks good to me

File: discord/app_commands/commands.py [hunk 1]
```diff
@@ -1065,6 +1065,9 @@ def autocomplete(
         The coroutine decorator **must** return a list of :class:`~discord.app_commands.Choice` objects.
         Only up to 25 objects are supported.
 
+        .. warning::
+            The results presented to the user are suggestions. The user may ignore them and input their own value.
```
Comment (Rapptz) [inline discord/app_commands/commands.py:1069]:
```suggestion
            The choices returned from this coroutine are suggestions. The user may ignore them and input their own value.
```

File: discord/app_commands/commands.py [hunk 2]
```diff
@@ -2271,6 +2274,9 @@ def autocomplete(**parameters: AutocompleteCallback[GroupT, ChoiceT]) -> Callabl
 
     For more information, see the :meth:`Command.autocomplete` documentation.
 
+    .. warning::
+        The results presented to the user are suggestions. The user may ignore them and input their own value.
```
Comment (Rapptz) [inline discord/app_commands/commands.py:2278]:
```suggestion
        The choices returned from this coroutine are suggestions. The user may ignore them and input their own value.
```

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: discord/app_commands/commands.py (hunk 2 above)
Comment (rdrescher909) [reply discord/app_commands/commands.py]: Implemented with latest commit

File: discord/app_commands/commands.py (hunk 1 above)
Comment (rdrescher909) [reply discord/app_commands/commands.py]: Implemented with latest commit.

**Review by Rapptz — APPROVED** [review]
Thanks.

**Pull request merged**